add trim token filter docs #8449 (#8461)
opensearch-project/documentation-website: _analyzers/token-filters/trim.md

---
layout: default
title: Trim
parent: Token filters
nav_order: 430
---

# Trim token filter

The `trim` token filter removes leading and trailing white space characters from tokens. 

> Many popular tokenizers, for example `standard`, `keyword`, and `whitespace` tokenizers, automatically strip any leading and trailing spaces during tokenization. When using these tokenizers, there is no need to configure an additional `trim` token filter. 

Reviewer: ```suggestion
Many popular tokenizers, such as `standard`, `keyword`, and `whitespace` tokenizers, automatically strip leading and trailing white space characters during tokenization. When using these tokenizers, there is no need to configure an additional `trim` token filter. 
```

{: .note}

## Example

> The following example request creates a new index named `my_pattern_trim_index` and configures an analyzer with a `trim` filter and a `pattern` tokenizer, which does not remove leading and trailing white space:

Reviewer: ```suggestion
The following example request creates a new index named `my_pattern_trim_index` and configures an analyzer with a `trim` filter and a `pattern` tokenizer, which does not remove leading and trailing white space characters:
```
